stream_alert/threat_intel_downloader/threat_stream.py

```diff
@@ -0,0 +1,273 @@
+                                 'reaches threshold %d. Stop retrieve more '
+                                 'IOCs.', self.threshold)
+                    continue_invoke = False
+                else:
+                    next_url = data['meta']['next']
+                    continue_invoke = True
+            elif https_req.status_code == 401:
+                raise ThreatStreamRequestsError('Response status code 401, unauthorized.')
+            elif https_req.status_code == 500:
+                raise ThreatStreamRequestsError('Response status code 500, retry now.')
+            else:
+                raise ThreatStreamRequestsError('Unknown status code {}, '
+                                                'do not retry.'.format(https_req.status_code)
+                                               )
+        except (requests.exceptions.Timeout,
+                requests.exceptions.ConnectionError,
+                requests.exceptions.ChunkedEncodingError):
+            LOGGER.exception('Caught other requests exceptions, retry now.')
+        return (intelligence, next_url, continue_invoke)
+
+    def runner(self, event):
+        """Process URL before making API call
+        Args:
+            event (dict): Contains lambda function invocation information. Initially,
+                Threat Intel Downloader lambda funciton is invoked by Cloudwatch
+                event. 'next_url' key will be inserted to event lambda function
+                invokes itself to retrieve more IOCs.
+
+        Returns:
+            (tuple): (list, str, bool)
+                - First object is a list of intelligence.
+                - Second object is a string of next token to retrieve more IOCs.
+                - Third object is bool to indicated if retrieve more IOCs from
+                    threat feed.
+        """
+        if not event:
+            return None, None, False
+
+        next_url = event.get(
+            'next_url',
+            '/api/v2/{}/?username={}&api_key={}&status={}&limit={}'.format(
+                self._API_RESOURCE,
+                self.api_user,
+                self.api_key,
+                self._IOC_STATUS,
+                self._API_MAX_LIMIT
+            )
+        )
+
+        if not next_url:
+            return None, None, False
+
+        return self._connect(next_url)
+
+    @staticmethod
+    def _epoch_time(time_str, days=90):
+        """Convert expiration time (in UTC) to epoch time
+        Args:
+            time_str (str): expiration time in string format
+                Example: '2017-12-19T04:45:18.412Z'
+            days (int): default expiration days which 90 days from now
+
+        Returns:
+            (int): Epoch time. If no expiration time presented, return to
+                default value which is current time + 90 days.
+        """
+
+        if not time_str:
+            return int((datetime.now()
+                        + timedelta(days)
```

Comment:
this is interesting formatting. pylint doesn't complain about this 🤔 ?

I'd suggest putting the `+` and `-` on the preceding line as such:
```
return int((datetime.now() +
            timedelta(days) -
            datetime.utcfromtimestamp(0)).total_seconds())
```

Reply:
[PEP8](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator) recommends to break lines before operants, but actually before and after operants are both okay if we have style consistently.

Reply:
Weird 🤔